app/src/main/java/io/homeassistant/companion/android/webview/WebViewActivity.kt

```diff
@@ -263,11 +259,11 @@ class WebViewActivity : BaseActivity(), io.homeassistant.companion.android.webvi
                     ) {
                         dispatchKeyEvent(KeyEvent(KeyEvent.ACTION_DOWN, KeyEvent.KEYCODE_E))
                     }
-                    return !unlocked
```

Comment:
Confused by why the `unlocked` status is returned here. 
If not required, not having a blocking call to `isAppLocked` here would be better.

Reply:
This function should return `true` if the gesture has been consumed/handled by the app, and `false` if it isn't (in which case it'll be passed back to the webview/system to handle). If the app is currently locked, it won't respond to the gesture, so it should return `false`.

The returned value is [used here](https://github.com/home-assistant/android/blob/cabd41e6d5f64fe5218dcbf6be2d6a838f7e6885/app/src/main/java/io/homeassistant/companion/android/util/OnSwipeListener.kt#L98), and [returned to the `View.OnTouchListener` here](https://github.com/home-assistant/android/blob/master/app/src/main/java/io/homeassistant/companion/android/util/OnSwipeListener.kt#L107), which corresponds to [the return value specified in the developer documentation here](https://developer.android.com/reference/kotlin/android/view/View.OnTouchListener?hl=en#onTouch(android.view.View,%20android.view.MotionEvent)).

Reply:
Thanks for clarifying the use of the unlock state here.

I had a gut feeling that calling `isAppLocked` here _every_ motion event could bog down the system / cause slow UI response, which is why I wanted to bring that up right away. Not sure if the function call would slow the system down too much though. 

What's your thoughts on the matter? Should we cache the unlock state in the WebViewActivity, or are the `isAppLocked` function call in these places not harmfull?

Reply:
Is there any reason why it shouldn't be cached? I don't think it is possible for this value to change between `onResume` and `onPause` aside from a lock callback, is there?

Reply:
Good point, those should indeed be the only places where the state could change.

I've made changes to keep the app lock state cached.